File: quick-attendance-api/models/user_type.ts
export const enum UserType {
  Owner = "Owner",
  Manager = "Manager",
  Member = "Member",
}

export function is_privileged_user_type(
  user_type: UserType | null | undefined,
) {
  if (
    user_type === null || user_type === undefined ||
    user_type === UserType.Member
  ) return false;
  return true;
}

File: client/lib/models/user_type.dart
enum UserType {
  owner("Owner"),
  manager("Manager"),
  member("Member"),
  unknown("Unknown");

  final String value;
  const UserType(this.value);

  static UserType from(String? code) {
    return UserType.values.firstWhere(
      (e) => e.value == code,
      orElse: () => UserType.unknown,
    );
  }
}
